packages/rs-drive-abci/src/execution/platform_events/block_start/patch_platform_version/mod.rs

```diff
@@ -0,0 +1,128 @@
+mod patch_0_15_example;
+
+use crate::error::Error;
+use crate::execution::platform_events::block_start::patch_platform_version::patch_0_15_example::patch_0_15_example;
+use dpp::block::block_info::BlockInfo;
+use dpp::prelude::BlockHeight;
+use dpp::util::deserializer::ProtocolVersion;
+use dpp::version::PlatformVersion;
+use indexmap::IndexMap;
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use std::ops::Range;
+use std::sync::RwLock;
+
+use crate::platform_types::platform_state::v0::PlatformStateV0Methods;
+use crate::platform_types::platform_state::PlatformState;
+
+static PREVIOUS_PATCH_PATTERN: RwLock<Option<(ProtocolVersion, Range<BlockHeight>)>> =
+    RwLock::new(None);
+
+type PatchFn = fn(PlatformVersion) -> PlatformVersion;
+
+type HeightToPatchRanges = IndexMap<Range<BlockHeight>, PatchFn>;
+
+static PATCHES: Lazy<HashMap<ProtocolVersion, HeightToPatchRanges>> = Lazy::new(|| {
+    HashMap::from_iter(vec![
+        (
+            1,
+            IndexMap::from_iter(vec![
+                (15..30, patch_0_15_example as PatchFn),
+                (30..45, patch_0_15_example as PatchFn),
+                // (45..BlockHeight::MAX, patch_0_15_example as PatchFn),
+            ]),
+        ),
+        // (
+        //     2,
+        //     IndexMap::from_iter(vec![(15..24, patch_0_15_example as PatchFn)]),
+        // ),
+    ])
+});
+
+/// Patch platform version to change function versions to fix chain halt bugs
+pub fn patch_platform_version(
+    block_info: &BlockInfo,
+    current_platform_version: &PlatformVersion,
+    block_platform_state: &mut PlatformState,
+) -> Result<(), Error> {
+    // Check if a patch that matches protocol version and block height is already applied
```

Comment:
We should be able to apply more than one patch.

Reply:
More than one patch for the same version? You can.

Reply:
Per block height

Reply:
Done